sdk/core/azure_core_tracing_opentelemetry/src/tracer.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use crate::span::{OpenTelemetrySpan, OpenTelemetrySpanKind};
+use azure_core::tracing::{SpanKind, Tracer};
+use opentelemetry::{
+    global::BoxedTracer,
+    trace::{TraceContextExt, Tracer as OpenTelemetryTracerTrait},
+    Context,
+};
+use std::sync::Arc;
+
+pub struct OpenTelemetryTracer {
+    inner: BoxedTracer,
+}
+
+impl OpenTelemetryTracer {
+    /// Creates a new OpenTelemetry tracer with the given inner tracer.
+    pub(super) fn new(tracer: BoxedTracer) -> Self {
+        Self { inner: tracer }
+    }
+}
+
+impl Tracer for OpenTelemetryTracer {
+    fn start_span(
+        &self,
+        name: &'static str,
```

Comment:
are all usages in Azure sdk okay to provide a `'static str`?

Reply:
For *keys*, why wouldn't they be? How often is a tracing key passed as some unknown `String` at runtime? At worst, these could be `Cow<'static, str>` but I think that's overengineering in any cases @LarryOsterman or I could come up with. Do you have experience otherwise?